stat_test.go

```diff
@@ -836,33 +837,89 @@ var linearRegressionTests = []struct {
 
 		tol: 1e-5,
 	},
+	{
+		name: "x, y length mismatch",
+
+		x:       make([]float64, 3),
+		y:       make([]float64, 2),
+		weights: make([]float64, 3),
+	},
+	{
+		name: "weights mismatch with x",
+
+		x:       make([]float64, 3),
+		y:       make([]float64, 3),
+		weights: make([]float64, 2),
+	},
+	{
+		name: "x, y length mismatch and weights nil",
+
+		x:       make([]float64, 3),
+		y:       make([]float64, 2),
+		weights: nil,
+	},
+	{
+		name:    "weights are nil",
+		x:       []float64{12.0, 24.0, 21.0},
+		y:       []float64{1.0, 2.0, 3.0},
+		weights: nil,
+	},
 }
 
 func TestLinearRegression(t *testing.T) {
 	for _, test := range linearRegressionTests {
-		alpha, beta := LinearRegression(test.x, test.y, test.weights, test.origin)
-		var r float64
-		if test.origin {
-			r = RNoughtSquared(test.x, test.y, test.weights, beta)
+		if strings.Contains(test.name, "mismatch") {
```

Comment:
@btracey I think it'd be more cumbersome, cause adding panics field for all of them, and then again running multiple tests inside the block with different `origin` and separate blocks for `LinearRegression`, `RSquared` and `RNoughtSquared`.

The current implementation just does a simple string match and goes ahead with the tests, which is a better implementation IMHO. What do you think?